### ISSUE TITLE
Existing Vehicles in Savegame not Updating

### ISSUE THREAD

**Issue by AKF2996** [loki79uk/FS25_UniversalAutoload#3]
First off Loki, thank you for all your hard work on what I believe is the most critical mod for the Farm Sim franchise. 

I manually created a package and was able to load and launch Universal Autoload into the game on one of my existing saves. Autoload is functioning well, albeit with some quirks, on newly purchased trailers and vehicles. However, existing vehicles and trailers in the savegame are not currently implementing autoload. Am I missing a setting/config or is this a bug. Ultimately not a big deal as I can just swap out the existing vehicles for new ones but thought I should raise the issue.

Again, thank you so much for all your work. I am super excited for all the new features!

**Comment by geeknessfr** [comment]
Hi,

As mentioned on the README page, the existing trailers don't have a saved configuration for this.

I just tested it, and all you need to do is modify the existing trailer at the workshop (where you repair your vehicles), near the shop dealer, or by placing one on your farm. In my game, I changed the rim color on the lowbed trailer, and after that, I was able to use the Autoloader.

Have fun ! 😉

**Comment by AKF2996** [comment]
Fantastic, thank you so much! I interpreted the README note as that being a bugged feature not currently available. Just tried it on the International 200 truck and it worked. Thank you so much!

**Comment by loki79uk** [comment]
I will work on the shop UI - it needs buttons like "apply", "save as default", etc..

For now, vehicle settings are only saved in your settings file when you actually buy one.  So you can edit in the workshop and it should work for the rest of that game, but you need to actually buy one to save the configuration as default for new vehicles.  So a workaround is to re-buy all of the vehicles you own, then don't save and reload the game.  Or you can do the same in a different savegame.

I didn't document any of this, as it is likely to change soon and then I will forget to change the docs..  There is certainly a discussion to be had about the expected behaviour for creating/updating new settings.

**Comment by geeknessfr** [comment]
Oh yes, as the settings are not saved individually "per savegame" but globally... we can create a game and buy all the trailers we use to have a default setting saved ! Nice, thank you !

**Comment by geeknessfr** [comment]
> I'd be interested to hear your thoughts (or anyone else) on what is the best default behaviour for this. I can save settings per vehicle, but it could get complicated to keep track across multiple savegames, and I think most people would be happy with just a single global setting. Buttons like "apply to all" and "set as default" would help, or I could also make a global option to set different behaviours for how to save. Needs some thought for sure!


I agree that a global setting per trailer would be sufficient. I can't currently think of a scenario where someone would need the same model of trailer to behave differently, _but I can imagine someone asking for that feature at some point ^^_ :)

I know it's possible for a mod to save files in the savegame folder, so maybe when purchasing a vehicle at the shop, it could start with a global configuration as a default behavior for the mod, without requiring any intervention to keep it simple. In that case, the settings could be saved in modSettings.

If needed, there could be a button to save a specific configuration for "this vehicle" for those who might need customized settings in certain situations. To avoid cluttering the XML file in modSettings, there could be another XML file saved in the savegame folder for fine-tuning settings for individual vehicles.

The idea would be to keep the modSettings XML "clean" and, more importantly, to avoid having leftover specific configurations in the global XML if a savegame is deleted.

--- 

Alternatively, another idea could be to have, for example, 3 to 5 preset slots for different configurations per vehicle. You could choose which one to apply when purchasing a vehicle or in the workshop.

Additionally, you could save a configuration to a specific slot if needed. If no slot is specified, it would default to saving a global configuration (e.g., in slot 0).
